training/app/controllers/concerns/maintenance.rb

```diff
@@ -0,0 +1,36 @@
+module Maintenance
+  extend ActiveSupport::Concern
+
+  MAINTENANCE_FILE = "#{Rails.root.join('tmp/maintenance/timing.txt')}"
+
+  included do
+    before_action :do_maintenance
+  end
+
+  def maintenance_mode?
+    File.exist?(MAINTENANCE_FILE)
+  end
+
+  def get_maintenance_period
+    maintenance_time = []
+    File.open(MAINTENANCE_FILE, 'r') do |file|
+      file.each_line do |line|
+        maintenance_time.push(line.split(' => '))
+      end
+    end
+
+    maintenance_time.to_h
+  end
+
+  def maintenance_period?
+    maintenance_time = get_maintenance_period
+    maintenance_time['start_at'].to_time <= DateTime.now && maintenance_time['end_at'].to_time > DateTime.now
```

Comment:
あれ、これで気づいたけど、 rubocopやfablicopは導入していないのか？(そっか。。ここまで誰も指摘していなかった。ｗ）

bundle exec rubocop training/app/controllers/concerns/maintenance.rb
すると、結構怒られますよ。(この部分でいうと `DateTime.now` )

一例:
```
app/controllers/concerns/maintenance.rb:4:22: C: Rails/FilePath: Please use Rails.root.join('path', 'to') instead.
  MAINTENANCE_FILE = "#{Rails.root.join('tmp/maintenance/timing.txt')}"
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/controllers/concerns/maintenance.rb:4:22: C: Style/MutableConstant: Freeze mutable objects assigned to constants.
  MAINTENANCE_FILE = "#{Rails.root.join('tmp/maintenance/timing.txt')}"
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/controllers/concerns/maintenance.rb:4:22: C: Style/UnneededInterpolation: Prefer to_s over string interpolation.
  MAINTENANCE_FILE = "#{Rails.root.join('tmp/maintenance/timing.txt')}"
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/controllers/concerns/maintenance.rb:4:25: C: Rails/FilePath: Please use Rails.root.join('path', 'to') instead.
  MAINTENANCE_FILE = "#{Rails.root.join('tmp/maintenance/timing.txt')}"
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/controllers/concerns/maintenance.rb:14:7: C: Naming/AccessorMethodName: Do not prefix reader method names with get_.
  def get_maintenance_period
      ^^^^^^^^^^^^^^^^^^^^^^
app/controllers/concerns/maintenance.rb:27:34: C: Rails/Date: Do not use to_time on Date objects, because they know nothing about the time zone in use.
    maintenance_time['start_at'].to_time <= DateTime.now && maintenance_time['end_at'].to_time > DateTime.now
                                 ^^^^^^^
app/controllers/concerns/maintenance.rb:27:45: C: Style/DateTime: Prefer Date or Time over DateTime.
    maintenance_time['start_at'].to_time <= DateTime.now && maintenance_time['end_at'].to_time > DateTime.now
                                            ^^^^^^^^^^^^
app/controllers/concerns/maintenance.rb:27:54: C: Rails/TimeZone: Do not use DateTime.now without zone. Use one of Time.zone.now, DateTime.current, DateTime.now.in_time_zone, DateTime.now.utc, DateTime.now.getlocal, DateTime.now.iso8601, DateTime.now.jisx0301, DateTime.now.rfc3339, DateTime.now.to_i, DateTime.now.to_f instead.
    maintenance_time['start_at'].to_time <= DateTime.now && maintenance_time['end_at'].to_time > DateTime.now
```

Reply:
gem 'rubocop'
bundle install
bundle exec rubocop

Reply:
すでに、fablicopは導入していたのですが実行していませんでした。
こちら、対応しました。